src/PdfSmith.BusinessLayer/Templating/HandlebarsTemplateEngine.cs

```diff
@@ -0,0 +1,89 @@
+using System.Collections.Concurrent;
+using System.Globalization;
+using HandlebarsDotNet;
+using PdfSmith.BusinessLayer.Exceptions;
+using PdfSmith.BusinessLayer.Services;
+
+namespace PdfSmith.BusinessLayer.Templating;
+
+public class HandlebarsTemplateEngine(TimeZoneTimeProvider timeZoneTimeProvider) : ITemplateEngine
+{
+    private readonly IHandlebars handlebars = Handlebars.Create();
+    private static readonly ConcurrentDictionary<string, HandlebarsTemplate<object, object>> TemplateCache = new();
+    private static TimeZoneTimeProvider? timeProvider;
```

Comment:
`TimeZoneTimeProvider` doesn't need to be static anymore, so delete this line and also line 24-25.